### PR TITLE
fix(curriculum): rework Project Euler 97

### PR DESCRIPTION
**Pull request by gikf**
- Reworks challenge to use argument in function call.
- Adds solution.
- Changes expected return type to string, otherwise for cases where 10th digit from the end is `0`, it wasn't possible to show all 10 last digits.
- Changes slightly challenge to allow adding more tests.
- Adds more tests. Primes for tests are taken from short list at https://primes.utm.edu/primes/download.php
- Changes function name to include word `large` instead of shortened `lrg`.
- Uses MathJax to improve look of math notation.
- Related to #40896.

Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

### CODE REVIEW

File: curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-97-large-non-mersenne-prime.md [hunk 1]
```diff
@@ -8,41 +8,81 @@ dashedName: problem-97-large-non-mersenne-prime
...
 # --description--
 
-The first known prime found to exceed one million digits was discovered in 1999, and is a Mersenne prime of the form 2<sup>6972593</sup>−1; it contains exactly 2,098,960 digits. Subsequently other Mersenne primes, of the form 2<sup><var>p</var></sup>−1, have been found which contain more digits.
+The first known prime found to exceed one million digits was discovered in 1999, and is a Mersenne prime of the form $2^{6972593} − 1$; it contains exactly 2,098,960 digits. Subsequently other Mersenne primes, of the form $2^p − 1$, have been found which contain more digits.
 
-However, in 2004 there was found a massive non-Mersenne prime which contains 2,357,207 digits: 28433×2<sup>7830457</sup>+1.
+However, in 2004 there was found a massive non-Mersenne prime which contains 2,357,207 digits: $28433 × 2^{7830457} + 1$.
 
-Find the last ten digits of this prime number.
+Find the last ten digits of non-Mersenne prime in the form $multiplier × 2^{power} + 1$.
```
Comment (nhcarrigan) [inline curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-97-large-non-mersenne-prime.md:15]:
```suggestion
Find the last ten digits of that non-Mersenne prime in the form $multiplier × 2^{power} + 1$.
```

File: curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-97-large-non-mersenne-prime.md [hunk 2]
```diff
@@ -8,41 +8,81 @@ dashedName: problem-97-large-non-mersenne-prime
...
-`lrgNonMersennePrime()` should return a number.
+`largeNonMersennePrime(19, 6833086)` should return a string.
 
 ```js
-assert(typeof lrgNonMersennePrime() === 'number');
+assert(typeof largeNonMersennePrime(19, 6833086) === 'string');
 ```
 
-`lrgNonMersennePrime()` should return 8739992577.
+`largeNonMersennePrime(19, 6833086)` should return string `3637590017`.
```
Comment (nhcarrigan) [inline curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-97-large-non-mersenne-prime.md:25]:
```suggestion
`largeNonMersennePrime(19, 6833086)` should return the string `3637590017`.
```

File: curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-97-large-non-mersenne-prime.md [hunk 3]
```diff
@@ -8,41 +8,81 @@ dashedName: problem-97-large-non-mersenne-prime
...
 
 ```js
-assert.strictEqual(lrgNonMersennePrime(), 8739992577);
+assert.strictEqual(largeNonMersennePrime(19, 6833086), '3637590017');
+```
+
+`largeNonMersennePrime(27, 7046834)` should return string `0130771969`.
```
Comment (nhcarrigan) [inline curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-97-large-non-mersenne-prime.md:31]:
```suggestion
`largeNonMersennePrime(27, 7046834)` should return the string `0130771969`.
```

File: curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-97-large-non-mersenne-prime.md [hunk 4]
```diff
@@ -8,41 +8,81 @@ dashedName: problem-97-large-non-mersenne-prime
...
+
+```js
+assert.strictEqual(largeNonMersennePrime(27, 7046834), '0130771969');
+```
+
+`largeNonMersennePrime(6679881, 6679881)` should return string `4455386113`.
```
Comment (nhcarrigan) [inline curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-97-large-non-mersenne-prime.md:37]:
```suggestion
`largeNonMersennePrime(6679881, 6679881)` should return the string `4455386113`.
```

File: curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-97-large-non-mersenne-prime.md [hunk 5]
```diff
@@ -8,41 +8,81 @@ dashedName: problem-97-large-non-mersenne-prime
...
+
+```js
+assert.strictEqual(largeNonMersennePrime(6679881, 6679881), '4455386113');
+```
+
+`largeNonMersennePrime(28433, 7830457)` should return string `8739992577`.
```
Comment (nhcarrigan) [inline curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-97-large-non-mersenne-prime.md:43]:
```suggestion
`largeNonMersennePrime(28433, 7830457)` should return the string `8739992577`.
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
Some nitpicks.

**Review by naomi-lgbt — APPROVED** [review]

**Review by Sembauke — APPROVED** [review]

**Comment by moT01** [comment]
I added this solution to the forum guide article

**Pull request merged**